src/els_text_edit.erl

```diff
@@ -0,0 +1,71 @@
+-module(els_text_edit).
+
+-export([ diff_files/2
+        ]).
+
+-include("erlang_ls.hrl").
+
+%%==============================================================================
+%% File and module functions
+%%==============================================================================
+
+%% @doc Construct a [text_edit()] from the diff between two files.
+%%
+%% Generate a list of text edits to transform `SourceFile' into
+%% `DestinationFile'
+-spec diff_files(els_uri:path(), els_uri:path()) -> [text_edit()].
+diff_files(SourceFile, DestinationFile) ->
+    Diffs = tdiff:diff_files(SourceFile, DestinationFile),
+    Patch = make_text_edits(Diffs),
+    Patch.
+
+
+%%==============================================================================
+%% Internal functions
+%%==============================================================================
+
+%% A text_edit() has a range and a blob of text.
+%%
+%% insert:
+%%    start and end of range are the same, specifying where to include
+%% delete:
+%%    start and end of range define region to be removed, blank text.
+%% change:
+%%    start and end are original range, text is replacement text
+
+-type diff() :: {del, [string()]} | {eq, [string()]} | {ins, [string()]}.
+
+-spec make_text_edits([diff()]) -> [text_edit()].
+make_text_edits(Diffs) ->
+    make_text_edits(Diffs, 0, []).
+
+-spec make_text_edits([diff()], number(), [text_edit()]) -> [text_edit()].
+make_text_edits([{eq, Data}|T], Line, Acc) ->
+    make_text_edits(T, Line + length(Data), Acc);
```

Comment:
Please make sure this works correctly with utf8 data. `lenth` on Erlang strings returns the number of code-points, which might or might not be what is expected here (vs byte size or grapheme count or utf-16 codepoint size).

Reply:
At this point the `Data` is a list of lines,  as we are operating at the whole line level.  We use the length of the list to advance the line counter.  So unicode encoding should not be a problem.